### ISSUE TITLE
sharkfin 12 V current limit should be decreased in software

### ISSUE THREAD

**Issue by wesolows** [oxidecomputer/hubris#1273]
The sharkfin rev C and newer schematics contain a note referring to the MAX5970 hot-swap power controller's channel 1 configuration:
```
Current Sense Range : 50mV
Maximum Current Sense Possible: 10A
Default Fast-Trip Current: 7.5A
Default Slow-Trip Current: 3.75A

FIRMWARE SETTINGS:
Desired Fast Trip Current: 6A (DAC_CH0: 0x99)
Desired Fast-to-Slow Ratio: 200% (Default)
Resulting Slow Trip Current: 3A
```
The MAX5970 doesn't, as far as we are told, have any firmware, but the intent is clear enough: ideally we should be setting the configuration registers from software to reduce this threshold, specifically the `dac_ch0` register referred to in the note.  It is possible (though unlikely, given the relatively small size of the change) that a faulty SSD or sharkfin could otherwise lead to preventable equipment damage in the case of an overcurrent fault.

If we want to do this, it would need to be done via I2C on all 10 installed fins prior to entering A1.  Fins are not hot-serviceable so it could be done any time during initial startup as long as it gates the first subsequent entry into A1.

**Comment by ericaasen** [comment]
For UV/OV detection, there is no requirement for 12V specified in SFF-8369, Table 4-1.  It specifically calls out that the tolerance on this rail is vendor specific.

The specifications for the WDC Ultrastar SN840 specify a voltage tolerance of +/-15% at the pin with a 2.45A maximum continuous current and a 4.5A peak current, which matches the SFF specification.  This is a range of 10.2-13.8V.

The setting below is something that I don't think we should hit, but should trip before the drives actually get unhappy.

### TLDR: What to set things to
```
Register       | Address | Value |
---------------+---------+-------|
UV1THR_CH1_MSB |    0x1A |  0xB3 |
UV1THR_CH1_LSB |    0x1B |  0x01 |
```
This sets the UV1THR register to `0x2CD` which sets a warning level at 11.21V.  

### How  I got this setting
The worst-case output voltage of the BMR491 is about 11.58V if you stack transient conditions of a 25% load jump along with full line and load regulation and initial accuracy from 50.25V-60V input and 0-100% output load.

The Gimlet has almost the entirety of layer 9 dedicated to 12V distribution and that is a 2oz layer, so the voltage drop across that from back to front of the Gimlet is assumed to be negligible at our relatively low loads, but will be assumed to be up to 10mV of drop.

There are 5 pins on the CEM slot that are used for 12V.  The LLCR (low-level contact resistance) of the CEM slot pins with a 1.6mm thick PCB is measured as 13-14 milliohm in the qualification test report by Samtec: https://suddendocs.samtec.com/testreports/1080554_report_rev_1%20-%2011232017_qua.pdf
Since there are 5 pins, that's 5 pins in parallel at 14 milliohms each or 2.8 milli-ohm of resistance, which is 12.6mV at 4.5A of current.

The current sense resistor on the 5970 is an additional 5 milli-ohms or 22.5 mV of drop at 4.5A.  The `MON` pin that actually measures the voltage on the 5970 is after this current sense resistor, so this must be taken into account.

If the worst-case of all of these is subtracted from the minimum output voltage of the BMR491

```
11.58
-0.01 (Gimlet planes)
-0.0126 (CEM connector)
-0.0225 (current sense resistor)
--------------------------
11.53V
```

The ADC is the 5970 (which is used to feed into the digital comparator that generates UV warnings and faults) is better than 2% accuracy based on the specified tolerance on LSB voltage that is 1.67% low, at worst, for the 16V range, so if we assume this ADC is accurate to no better than 2% of Full-Scale Range, we should be conservative.  2% of 16V is 0.32V, so we will want to set the UV threshold 0.32V lower than we actually want it to guarantee we don't accidentally trip the warning.  Even though this seems close to the specified 1.67%, we also have other tolerances all stacked up as worst-case, which generally never happens, so this is fine.

If we wanted to have no false warnings, we should guarantee that we don't set the UV warning at an actual voltage of 11.53V or above.  To achieve this, we need to set the ADC to 11.53V - 0.32V or 11.21V.  This will give us an `ALERT_L` when the voltage is lower than it should ever be under normal operating conditions.

<img width="506" alt="image" src="https://github.com/oxidecomputer/hubris/assets/83726929/b72d3deb-c671-4da6-a3e3-042f38f0048b">
